Composer/packages/server/src/models/connector/csharpBotConnector.ts

```diff
@@ -2,74 +2,142 @@
+        build.stderr.on('data', function(err) {
+          reject(err.toString());
+        });
+
+      build.on('exit', function(code) {
+        resolve(code);
+      });
+    });
+  };
+
+  private getConnectorConfig = (config: DialogSetting) => {
+    const configList: string[] = [];
+    if (config.MicrosoftAppPassword) {
+      configList.push('--MicrosoftAppPassword');
+      configList.push(config.MicrosoftAppPassword);
+    }
+    if (config.luis) {
+      if (config.luis.authoringKey) {
+        configList.push('--luis:endpointKey');
+        configList.push(config.luis.authoringKey);
+      }
+      if (config.luis.authoringRegion) {
+        configList.push('--luis:endpoint');
+        configList.push(`https://${config.luis.authoringRegion}.api.cognitive.microsoft.com`);
+      }
     }
 
-    this.status = BotStatus.NotConnected;
+    return configList;
+  };
+  private addListeners = (child: ChildProcess, handler: Function) => {
+    if (child.stdout !== null) {
+      child.stdout.on('data', (data: any) => {
+        console.log(`stdout: ${data}`);
+      });
+    }
+
+    if (child.stderr !== null) {
+      child.stderr.on('data', (data: any) => {
+        console.log(`stderr: ${data}`);
+      });
+    }
+
+    child.on('close', code => {
+      console.log(`close ${code}`);
+      handler();
+    });
+
+    child.on('error', (err: any) => {
+      console.log(`stderr: ${err}`);
+    });
+
+    child.on('exit', code => {
+      console.log(`exit: ${code}`);
+      handler();
+    });
 
-    return `${this.endpoint}/api/messages`;
+    child.on('message', msg => {
+      console.log(msg);
+    });
+
+    child.on('disconnect', code => {
+      console.log(`disconnect: ${code}`);
+      handler();
+    });
   };
 
-  sync = async (config: DialogSetting) => {
-    // archive the project
-    // send to bot runtime service
+  private getBotPath = () => {
     const currentProject = BotProjectService.getCurrentBotProject();
     if (currentProject === undefined) {
       throw new Error('no project is opened, nothing to sync');
     }
-    const dir = Path.join(currentProject.dataDir);
-    const luisConfig = currentProject.luPublisher.getLuisConfig();
-    await this.archiveDirectory(dir, './tmp.zip');
-    const content = fs.readFileSync('./tmp.zip');
-
-    const form = new FormData();
-    form.append('file', content, 'bot.zip');
+    return Path.join(currentProject.dir);
+  };
 
-    if (luisConfig && luisConfig.authoringKey !== null && !currentProject.checkLuisPublished()) {
-      throw new Error('Please publish your Luis models');
-    }
+  private start = async (dir: string, config: DialogSetting) => {
+    runtime = spawn(
+      'dotnet',
+      ['bin/Debug/netcoreapp2.1/BotProject.dll', `--urls`, this.endpoint, ...this.getConnectorConfig(config)],
+      {
+        detached: true,
+        cwd: dir,
+        stdio: ['ignore', 'ignore', 'inherit'],
+      }
+    );
+    console.log(`start runtime at ${runtime.pid}`);
```

Comment:
debug logger here.

Copy should be something like:

`runtime started. pid: ${runtime.pid}`